### PR TITLE
fix: correctness fixes from super-qa #505 (medium)

### PR DESCRIPTION
**Pull request by dutiona**
Four **behavior-correctness** fixes from the super-qa #505 medium sweep, each implemented TDD-first (failing test → fix → green).

### Findings
- **bootstrap** (`extract.rs`): `KeywordExtractor` left `metadata.session_outcome` **absent** for `Indeterminate` sessions (Success/Failure wrote `"success"`/`"failure"`). Collapsed the match so it now writes `"indeterminate"`. *Test:* `keyword_extractor_indeterminate_sets_outcome_key`.
- **cli** (`stats.rs`): `stats --format plain` omitted `scopes.max_depth` and `storage.page_count` that the Table format includes. Added both lines (parity). *Test:* `stats_plain_includes_max_depth_and_page_count`.
- **mcp** (`tools/mod.rs`, `memory_query`): an unknown `scope_mode` silently fell back to `subtree`. Now returns `invalid_params` ("unknown scope_mode: …") per the `parse_search_mode` convention; an **omitted** `scope_mode` still defaults to `subtree`. *Test:* `query_with_invalid_scope_mode_returns_error`.
- **store** (`facts.rs`, `next_due_time`): a corrupt stored `t_valid` was mapped to `MemoryError::Migration` ("schema migration failed") — wrong variant. Now uses the canonical `parse_timestamp` helper → `MemoryError::Database` (TEXT→timestamp conversion), matching `stamp_surfaced`/`row_to_fact`. *Test:* `next_due_time_corrupt_t_valid_is_database_error_not_migration`.

### Notes
- **Data forward-compat** (documented in ADR-0009): pre-fix DBs have `session_outcome` absent for Indeterminate facts; SQL consumers must treat a missing `$.session_outcome` as "unknown/legacy", not a specific outcome.
- **CLI edge case** (intended): an explicit empty-string `scope_mode` now errors instead of defaulting — strict validation against the schema enum.

### Verification
Gate: `cargo build --workspace` ✅, `cargo test --workspace` ✅, `cargo clippy --workspace --all-targets` ✅, `cargo fmt --check` ✅. Adversarial review (3 lenses): correctness ✅, no-regression ✅, api-and-persist ✅ (nit folded in).

Part of #505.

### CODE REVIEW

File: src/store/facts.rs [hunk 1]
```diff
@@ -430,12 +430,7 @@ impl<'a> FactStore<'a> {
         let result: Option<String> =
             stmt.query_row(rusqlite::params_from_iter(params), |r| r.get(0))?;
         match result {
-            Some(s) => {
-                let dt = DateTime::parse_from_rfc3339(&s)
-                    .map_err(|e| crate::error::MemoryError::Migration(format!("bad t_valid: {e}")))?
-                    .with_timezone(&Utc);
-                Ok(Some(dt))
-            }
+            Some(s) => Ok(Some(crate::store::parse_timestamp(&s)?)),
```
[the remark this answers is not in the document]
Comment (dutiona) [reply src/store/facts.rs]: Done in 48bdf76 — using the already-imported `parse_timestamp` (dropped the redundant `crate::store::` path).

File: src/store/facts.rs [hunk 2]
```diff
@@ -1054,6 +1049,36 @@ mod tests {
...
     }
 
+    #[test]
+    fn next_due_time_corrupt_t_valid_is_database_error_not_migration() {
+        let conn = setup();
+        let fs = FactStore::new(&conn, DIM);
+        let now = Utc::now();
+
+        let mut future = make_fact("reminder", vec![0.1; DIM]);
+        future.t_valid = Some(now + chrono::Duration::hours(1));
```
[the remark this answers is not in the document]
Comment (dutiona) [reply src/store/facts.rs:1060]: Done in 48bdf76 — `TimeDelta::hours(VALID_DURATION_HOURS)` with a named const, matching the rest of the suite.
